src/platform/forms-system/test/js/fields/FileField.unit.spec.jsx

```diff
@@ -919,4 +920,426 @@ describe('Schemaform <FileField>', () => {
+        },
+      ];
+      const errorSchema = {
+        0: {
+          __errors: ['Bad error'],
+        },
+      };
+      const registry = {
+        fields: {
+          SchemaField: f => f,
+        },
+      };
+      const tree = shallow(
+        <FileField
+          registry={registry}
+          schema={schema}
+          uiSchema={uiSchema}
+          idSchema={idSchema}
+          errorSchema={errorSchema}
+          formData={formData}
+          formContext={formContext}
+          onChange={f => f}
+          requiredSchema={requiredSchema}
+          enableShortWorkflow
+        />,
+      );
+
+      const cancelButton = tree.find('button.usa-button-secondary');
+      expect(cancelButton.text()).to.equal('Cancel');
+      tree.unmount();
+    });
+
+    it('should render delete button', () => {
+      const idSchema = {
+        $id: 'field',
+      };
+      const schema = {
+        additionalItems: {},
+        items: [
+          {
+            properties: {},
+          },
+        ],
+      };
+      const uiSchema = fileUploadUI('Files');
+      const formData = [
+        {
+          uploading: false,
+        },
+      ];
+      const registry = {
+        fields: {
+          SchemaField: f => f,
+        },
+      };
+      const tree = shallow(
+        <FileField
+          registry={registry}
+          schema={schema}
+          uiSchema={uiSchema}
+          idSchema={idSchema}
+          formData={formData}
+          formContext={formContext}
+          onChange={f => f}
+          requiredSchema={requiredSchema}
+          enableShortWorkflow
+        />,
+      );
+
+      const cancelButton = tree.find('button.usa-button-secondary');
+      expect(cancelButton.text()).to.equal('Delete file');
+      tree.unmount();
+    });
+  });
+
+  describe('enableShortWorkflow is false', () => {
+    it('should not render upload button while file has error', () => {
+      const idSchema = {
+        $id: 'field',
+      };
+      const schema = {
+        additionalItems: {},
+        items: [
+          {
+            properties: {},
+          },
+        ],
+        maxItems: 4,
+      };
+      const uiSchema = fileUploadUI('Files');
+      const formData = [
+        {
+          errorMessage: 'asdfas',
+        },
+      ];
+      const errorSchema = {
+        0: {
+          __errors: ['ERROR-123'],
+        },
+      };
+      const registry = {
+        fields: {
+          SchemaField: f => f,
+        },
+      };
+      const tree = shallow(
+        <FileField
+          registry={registry}
+          schema={schema}
+          uiSchema={uiSchema}
+          idSchema={idSchema}
+          errorSchema={errorSchema}
+          formData={formData}
+          formContext={formContext}
+          onChange={f => f}
+          requiredSchema={requiredSchema}
+        />,
+      );
+
+      const uploadButton = tree.find('#field_add_label');
+      expect(uploadButton.exists()).to.be.true;
```

Comment:
Same comment here about the selector - I'm not sure what it's referring to.

Reply:
Added a comment to help discern that the id is interpolated.